code/modules/mentor/mentor.dm

```diff
@@ -0,0 +1,94 @@
+GLOBAL_LIST_EMPTY(mentor_datums)
+GLOBAL_PROTECT(mentor_datums)
+
+GLOBAL_VAR_INIT(mentor_href_token, GenerateToken())
+GLOBAL_PROTECT(mentor_href_token)
+
+/datum/mentors
+	var/name = "someone's mentor datum"
+	/// The actual mentor, client type.
+	var/client/owner
+	/// The Mentor's ckey.
+	var/target
+	/// href token for mentor commands, uses the same token used by admins.
+	var/href_token
+	var/mob/following
+
+/datum/mentors/New(ckey)
+	if(!ckey)
+		QDEL_IN(src, 0)
+		CRASH("Mentor datum created without a ckey")
+	target = ckey(ckey)
+	name = "[ckey]'s mentor datum"
+	href_token = GenerateToken()
+	GLOB.mentor_datums[target] = src
+	// Set the owner var and load commands.
+	owner = GLOB.directory[ckey]
+	if(owner)
+		owner.mentor_datum = src
+		owner.add_mentor_verbs()
+		if(!check_rights_for(owner, R_ADMIN,0)) // Don't add admins to mentor list.
+			GLOB.mentors += owner
+
+/datum/mentors/proc/CheckMentorHREF(href, href_list)
+	return TRUE
+	/*			Disabled
```

Comment:
do you plan on implementing this later?

Reply:
CheckAdminHref does the same right now, this will be done in a different PR